proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -925,9 +931,29 @@ def initialize(blk, initialize_guess=None, state_args=None, outlvl=idaeslog.NOTS
             solver : solver object or string indicating which solver to use during
                      initialization, if None provided the default solver will be used
                      (default = None)
+            fail_on_warning : boolean argument to fail or only produce  warning upon unsuccessful solve (default=False)
+
         Returns:
             None
         """
+        def _check_solve(results, checkpoint=None):
+            def _raise_if_infeasible(fail_flag):
+                msg = f"{checkpoint} failed. The solver failed to converge to an optimal solution. " \
+                      f"This suggests that the user provided infeasible inputs or that the model is poorly scaled."
+                if fail_flag is True:
+                    raise RuntimeError(msg)
+                elif fail_flag is False:
+                    init_log.warning(msg)
+                else:
+                    raise Exception(f'The fail_on_warning argument in the initialize method was set to {fail_flag}. '
+                                    f'fail_on_warning is a boolean argument. Set fail_on_warning to True or False.')
+            if checkpoint is None:
+                checkpoint = 'Initialization step'
+            if check_optimal_termination(results):
+                init_log.info(f'{checkpoint} successful.')
```

Comment:
Your logger level here differs from that of the original property package, and assumes `info` is always the desired level.

It might be worth making the logger level and optional argument as well.

Reply:
Admittedly, I don't know too much about the different logger levels. I'll look into this.

Reply:
First, note that this only really applies if we are making this a general utility tool. For custom property packages, do whatever you want.

All you really need is a new argument at the top `loglvl=idaeslog.info` and then use `loglvl` wherever you are logging a message.

This will default to the `info` level, but allow users of the method to select a different logger level if desired to match their needs.